Fix interface lookup

thesefer opened the pull request:
## Description
<!--- Describe your changes in detail, what problems does it solve? -->
In https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/blob/master/templates/etc/mysql/conf.d/galera.cnf.j2 the subvalue of `hostvars[node]['ansible_' ~ galera_cluster_bind_interface]['ipv4']['address']` cannot be found because the `galera_cluster_bind_interface` contains only the interface for the current host. You have to lookup the variable for each host individually.

EDIT: 
To be more precise, in it's old form running the template on host 1, the `galera_cluster_bind_interface` would be `eth0` while on the second host it is `eth1`. The check running on host1 would check `hostvars[host1]['ansible_' ~ eth0]['ipv4']['address']` and gets a result but would not get anything for `hostvars[host2]['ansible_' ~ eth0]['ipv4']['address']` because eth0 does not exist there. Same vice versa. The `galera_cluster_bind_interface` is set fixed for the host it is running on.

## Related Issue
<!--- If you are suggesting a new feature or change, please create an issue first -->
<!--- Please link to the issue, discourse, or stackoverflow here: -->
https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/issues/243

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
Couldn't test properly due to company proxy restrictions.

thesefer committed:
Fix interface lookup

In https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/blob/master/templates/etc/mysql/conf.d/galera.cnf.j2 the subvalue of `hostvars[node]['ansible_' ~ galera_cluster_bind_interface]['ipv4']['address']` cannot be found because the `galera_cluster_bind_interface` contains only the interface for the current host. You have to lookup the variable for each host individually.

eRadical commented:
This means on each node one has to set "galera_cluster_bind_interface".
I find it rather cumbersome...

I'm also not sure if this is backward compatible.

thesefer commented:
I didn't check or change the following occurrences:

```
roles/mrlesmithjr.mariadb-galera-cluster/templates/etc/my.cnf.d/server.cnf.j2
51:{%   set _ = _galera_cluster_node_addresses.append( hostvars[node]['galera_cluster_bind_address'] | default(hostvars[node]['ansible_' ~ galera_cluster_bind_interface]['ipv4']['address']) | mandatory ) %}

roles/mrlesmithjr.mariadb-galera-cluster/templates/etc/my.cnf.d/server.cnf.temp.j2
51:{%   set _ = _galera_cluster_node_addresses.append( hostvars[node]['galera_cluster_bind_address'] | default(hostvars[node]['ansible_' ~ galera_cluster_bind_interface]['ipv4']['address']) | mandatory ) %}

roles/mrlesmithjr.mariadb-galera-cluster/templates/etc/mysql/conf.d/galera.cnf.temp.j2
51:{%   set _ = _galera_cluster_node_addresses.append( hostvars[node]['galera_cluster_bind_address'] | default(hostvars[node]['ansible_' ~ galera_cluster_bind_interface]['ipv4']['address']) | mandatory ) %}
``` 

@eRadical 
The default is `galera_cluster_bind_interface: "eth0"` which wouldn't match either so one **has** to set it. I worked around this by setting it to the following: `galera_cluster_bind_interface: "{{ ansible_default_ipv4['interface'] }}"` in my group vars.

thesefer commented:
Btw the following default variable isn't working / expanding and needs to be changed according to my tests but is not covered in this PR:

`galera_cluster_bind_address: "{{ hostvars[inventory_hostname]['ansible_' + galera_cluster_bind_interface]['ipv4']['address'] }}"` to `galera_cluster_bind_address: "{{ ansible_default_ipv4['address'] }}"`

thesefer committed:
Update main.yml

thesefer committed:
Update server.cnf.j2

thesefer committed:
Update server.cnf.temp.j2

thesefer committed:
Update galera.cnf.temp.j2

thesefer commented:
I implemented the change to all files considered. I added the example for `galera_cluster_bind_interface` to the defaults file (which will make it completely dynamic because it is using ansible runtime variables and values) and replaced the `+galera_cluster_bind_address` to do the same.

Please note that the fix from https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/commit/0d3180f36a4f78fbbb8cc0f2f035b8457c6f1292 will not work (Please test it by removing the default completely), you will only get `The error was: ansible.errors.AnsibleFilterError: Mandatory variable '10.10.10.0'  not defined` because it is using the value of the variable.

If one didn't do some weird magic this should be 100% backwards compatible (with all fixes applied) because it is using exactly the same values which ansible would gather anyways.

eRadical commented:
Hi @thesefer - please rebase & resolve conflicts so I can give it a run.

thesefer committed:
Merge branch 'master' into patch-galera-template

thesefer commented:
Hi @eRadical. Conflicts resolved.

thesefer committed:
Update converge.yml

make molecule aware of it's defaults

thesefer committed:
Update side_effect.yml

make molecule aware of its defaults

thesefer commented:
Please don't close

eRadical commented:
Keep

eRadical commented:
keep

thesefer committed:
Merge branch 'mrlesmithjr:master' into patch-galera-template
